web-client/src/views/Messages/MessageTable.tsx

```diff
@@ -0,0 +1,408 @@
+                      columnData={columnData}
+                      formattedMessages={formattedMessages}
+                      key={columnData.sortFieldInfo.sortField}
+                      messageListId={id}
+                      tableSort={tableSort}
+                      onSort={sortTableSequence}
+                    />
+                  );
+                })}
+              </tr>
+            </thead>
+            {formattedMessages.messages.map(message => {
+              return (
+                <MessageRow
+                  columns={messageColumns}
+                  key={message.messageId}
+                  message={message}
+                  messageListId={id}
+                  selectable={selectable}
+                  onSelect={setSelectedMessagesSequence}
+                />
+              );
+            })}
+          </table>
+          {!formattedMessages.hasMessages && <div>There are no messages.</div>}
+        </div>
+      </>
+    );
+  },
+);
+
+const MessageColumnHeader = ({
+  columnData,
+  formattedMessages,
+  messageListId,
+  onSort,
+  tableSort,
+}: {
+  messageListId: string;
+  columnData: MessageColumnData;
+  tableSort: any;
+  formattedMessages: any;
+  onSort: ({
+    sortField,
+    sortOrder,
+  }: {
+    sortField: string;
+    sortOrder: 'asc' | 'desc';
+  }) => void;
+}) => {
+  return (
+    <>
+      {columnData.headerIconClassName && (
+        <th className={columnData.headerIconClassName}></th>
+      )}
+      <th
+        aria-label={columnData.columnName}
+        className={columnData.headerClassName}
+        colSpan={
+          columnData.sortFieldInfo === SORT_FIELDS.DOCKET_NUMBER ? 2 : undefined // Takes into account the consolidated case icon
+        }
+        // TODO: probably should use aria-sort, but USWDS has default styles for this we may not want
+      >
+        <SortableColumn
+          ascText={getAscendingTextForSortType(
+            columnData.sortFieldInfo.sortType,
+          )}
+          currentlySortedField={tableSort.sortField}
+          currentlySortedOrder={tableSort.sortOrder}
+          data-testid={`${messageListId}-${columnData.sortFieldInfo.sortField}-header-button`}
+          defaultSortOrder={
+            columnData.sortFieldInfo === SORT_FIELDS.DOCKET_NUMBER
+              ? DESCENDING
+              : ASCENDING
+          }
+          descText={getDescendingTextForSortType(
+            columnData.sortFieldInfo.sortType,
+          )}
+          hasRows={formattedMessages.hasMessages}
+          sortField={columnData.sortFieldInfo.sortField}
+          title={columnData.columnName}
+          onClickSequence={onSort}
+        />
+      </th>
+    </>
+  );
+};
+
+const MessageRow = ({
+  columns,
+  message,
+  messageListId,
+  onSelect,
+  selectable,
+}: {
+  message: any;
+  messageListId: string;
+  selectable: boolean;
+  onSelect: (messages: any) => void;
+  columns: MessageColumnData[];
+}) => {
+  return (
+    <tbody>
```

Comment:
I'm sure this was pre-existing, but do we really want a `<tbody>` element for every row?

Reply:
Thanks for catching this! I'm slipping the fix into another PR.